src/briefcase/config.py

```diff
@@ -8,14 +8,125 @@
 
 from .exceptions import BriefcaseConfigError
 
-# The restriction on application naming comes from PEP508
+# The first restriction on application naming comes from PEP508
 PEP508_NAME_RE = re.compile(
     r'^([A-Z0-9]|[A-Z0-9][A-Z0-9._-]*[A-Z0-9])$',
     re.IGNORECASE
 )
 
-# This is the canonical definition from PEP440, modified to include
-# named groups
+# Create a list of reserved keywords for Python, Javascript, Java, C, and C++
+RESERVED_KEYWORDS = [
+    '_Packed',
+    'abstract',
+    'and',
+    'as',
+    'assert',
+    'auto',
+    'boolean',
+    'break',
+    'byte',
+    'case',
+    'catch',
+    'char',
+    'class',
+    'const',
+    'continue',
+    'debugger',
+    'def',
+    'default',
+    'del',
+    'delete',
+    'do',
+    'double',
+    'elif',
+    'else',
+    'enum',
+    'except',
+    'exec',
+    'export',
+    'extends',
+    'extern',
+    'false',
+    'final',
+    'finally',
+    'float',
+    'for',
+    'from',
+    'function',
+    'global',
+    'goto',
+    'if',
+    'implements',
+    'import',
+    'in',
+    'instanceof',
+    'int',
+    'interface',
+    'is',
+    'lambda',
+    'long',
+    'native',
+    'new',
+    'None',
+    'nonlocal',
+    'not',
+    'or',
+    'package',
+    'pass',
+    'print',
+    'private',
+    'protected',
+    'public',
+    'raise',
+    'register',
+    'return',
+    'short',
+    'signed',
+    'sizeof',
+    'static',
+    'strictfp',
+    'struct',
+    'super',
+    'switch',
+    'synchronized',
+    'this',
+    'throw',
+    'throws',
+    'transient',
+    'true',
+    'try',
+    'typedef',
+    'typeof',
+    'union',
+    'unsigned',
+    'var',
+    'void',
+    'volatile',
+    'while',
+    'with',
+    'yield',
+]
+
+
+def is_valid_pep508_name(app_name):
+    if PEP508_NAME_RE.match(app_name):
+        return True
+    return False
```

Comment:
The logic here is correct, but a little more complex than it needs to be. The 2 lines can be simplified to 
`return PEP505_NAME_RE.match(app_name)`